### PR TITLE
New terrain generator

### PR DESCRIPTION
**Pull request by creatorfromhell**
- Includes biome support
- Deterministic
- Includes various decorators(flowers, tall grass, ferns, multiple tree types, cacti, ores, water, snow/frozen water, and sugarcane)
- Things I would like to complete post-PR include overhangs, caves, improved terrain generation & biome transitions, and biome assigning according to elevation.

### CODE REVIEW

File: TrueCraft/Program.cs [hunk 1]
```diff
@@ -61,7 +64,11 @@ static void HandleChatMessageReceived(object sender, ChatMessageEventArgs e)
             {
                 e.PreventDefault = true;
                 var messageArray = e.Message.TrimStart('/')
+<<<<<<< HEAD
                     .Split(new[] {' '}, StringSplitOptions.RemoveEmptyEntries);
+=======
+                .Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+>>>>>>> New terrain generator
                 CommandManager.HandleCommand(e.Client, messageArray[0], messageArray);
```
Comment (SirCmpwn) [inline TrueCraft/Program.cs]: Bad merge here

File: TrueCraft/Program.cs [hunk 2]
```diff
@@ -42,6 +42,9 @@ public static void Main(string[] args)
             #if DEBUG
             server.AddLogProvider(new FileLogProvider(new StreamWriter("packets.log", false), LogCategory.Packets));
             #endif
+            #if DEBUG
+            server.AddLogProvider(new FileLogProvider(new StreamWriter("packets.log", false), LogCategory.Packets));
+            #endif
```
Comment (SirCmpwn) [inline TrueCraft/Program.cs]: Duplicated line here, probably bad merge

File: TrueCraft.Core/Logic/Items/FlintAndSteelItem.cs [hunk 3]
```diff
@@ -1,43 +0,0 @@
...
-
-        public bool SignificantMetadata
-        {
-            get
-            {
-                return false;
-            }
-        }
-    }
-}
```
Comment (SirCmpwn) [inline TrueCraft.Core/Logic/Items/FlintAndSteelItem.cs]: Why did you delete this file?

File: TrueCraft.API/TrueCraft.API.csproj [hunk 4]
```diff
@@ -31,6 +31,9 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="Coordinates3D.cs" />
+    <Compile Include="ArmourMaterial.cs" />
```
Comment (SirCmpwn) [inline TrueCraft.API/TrueCraft.API.csproj]: This is called `ArmorMaterial` now, and the project doesn't compile like this.

**Comment by SirCmpwn** [comment]
Handful of problems that would prevent a clean merge. I'll give some feedback on the actual terrain generation now.

**Comment by SirCmpwn** [comment]
We can probably merge this with these concerns outstanding and fix them over time.

* Trees are great
  * Good density in most biomes, but the snowy forest one is a bit too dense
* Cactus density is too high
* Biomes should probably be 10x bigger
* Seeing a lot of "desert next to tundra" effect
* Low res noise could be more extreme and lower res
* More sand before you get to sandstone in deserts
* Clay decorator
* Snow on water should probably only go so far so that sometimes we have ice exposed to the air

**Comment by SirCmpwn** [comment]
Also, ores don't seem to be there. I'd like to see that fixed before this is merged.

**Comment by creatorfromhell** [comment]
For my own reference.
* Trees in taiga are a bit too dense - fixed
* Cactus density is too high - fixed
* Biomes should probably be 10x bigger - fixed
* Seeing a lot of "desert next to tundra" effect
* Low res noise could be more extreme and lower res - fixed
* More sand before you get to sandstone in deserts - fixed
* Clay decorator
* Snow on water should probably only go so far so that sometimes we have ice exposed to the air - fixed
* Ores are not generating for some reason

**Comment by SirCmpwn** [comment]
* [ ] Trees are too dense in taigas
* [ ] Cactus density is too high
* [ ] Bimoes should be 10x bigger
* [ ] Desert appearing next to tundra
* [ ] Low res noise could be more extreme and lower res
* [ ] More sand before you get to sandstone in deserts
* [ ] Clay decorator
* [ ] Snow on water should probably only go so far so that we have ice exposed to air
* [ ] Ores are not generating
